aws/amazon-chime-sdk-component-library-react
# Add Observer to select input device error

**Issue #433 :** 

**Description of changes:**
Fix several typos. 
Add observer to `selectAudioInputDeviceError` and `selectVideoInputDeviceError` to deliver the error from SDK level to client application level.

When calling the `meetingManager.selectVideoInputDevice()/selectAudioInputDevice()`
* If there is an error, assign the error to `selectVideoInputDeviceError`/`selectAudioInputDeviceError`, and publish it
* If there is no error, set the `selectVideoInputDeviceError`/`selectAudioInputDeviceError` to `null`, and publish it

Error can be accessed through `useVideoInput`/`useAudioInput` hook. 

**Testing**
1. Have you successfully run `npm run build:release` locally?
    Yes

2. How did you test these changes?
    Mock a non-existent device for both `VideoInputProvider` and `AudioInputProvider`, and let `CameraSelection` and `MicSelection` display the `selectDeviceError` if not `null`, hide it if `null`.
    Run the meeting demo, on the "Device" page, for both video input device and audio input device:
    * Select the mocked device, and then the error appears
    * Choose the correct device, the error disappears
 
    Test on Safari, Chrome, Firefox.
3. If you made changes to the component library, have you provided corresponding documentation changes?
Yes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
